pallets/transfer-allowlist/src/lib.rs

```diff
@@ -0,0 +1,565 @@
+// Copyright 2023 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+/// This module checks whether an account should be allowed to make a transfer to
+/// a receiving location with a specific currency.
+/// If there are no allowances specified, then the account is assumed to be allowed
+/// to send to any location without restrictions.
+/// However once an allowance for a sender to a specific recieving location and currency is made,
+/// /then/ transfers from the sending account are restricted for that currency to:
+/// - the account(s) for which allowances have been made
+/// - the block range specified in the allowance
+use codec::{Decode, Encode, MaxEncodedLen};
+use frame_support::{dispatch::DispatchError, RuntimeDebugNoBound};
+pub use pallet::*;
+use pallet_connectors::DomainAddress;
```

Comment:
@NunoAlexandre If this type is used here, could it be moved to `cfg-types` to avoid the connector dependency?

Reply:
@lemunozm yeah I think that's a good idea. Probably worth doing that in this PR, wdyt?

Reply:
I agree!

Reply:
Thanks Thea :)

Reply:
No problem :)